src/components/App.js

```diff
@@ -0,0 +1,118 @@
+import Navigation from './Navigation'
+import Form from './Form'
+import Main from './Main'
+import Submit from './Submit'
+
+import '../css/oswald.css'
+import '../css/open-sans.css'
+import '../css/pure-min.css'
+import './App.css'
+
+class App extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      web3: null,
+      balance: 0,
+      currentAddress: null,
+      sendAddress: null,
+      sendAmount: null
+    }
+  }
+
+  componentDidMount() {
+    // Get network provider and web3 instance.
+    // See utils/getWeb3 for more info.
+    this.timeout = setInterval(this.pollWeb3, 1500)
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timeout)
+  }
+
+  pollWeb3 = () => {
+    getWeb3
+      .then(results => {
+        this.setState({
+          web3: results.web3
+        })
+
+        // Instantiate contract once web3 provided.
+        this.instantiateContract()
+      })
+      .catch(() => {
+        console.log('Error finding web3.')
+      })
+  }
+
+  instantiateContract() {
+    const munCoin = contract(MUNCoinContract)
+    munCoin.setProvider(this.state.web3.currentProvider)
+
+    var munCoinInstance
+
+    this.state.web3.eth.getAccounts((error, accounts) => {
+      munCoin.deployed().then((instance) => {
+        munCoinInstance = instance
+
+        // add contract instance to state
+        return this.setState({munCoinInstance: instance})
+      }).then((result) => {
+        // Get the value from the contract to prove it worked.
+        return munCoinInstance.balanceOf.call(accounts[0])
+      }).then((result) => {
+        // Update state with the result.
+        return this.setState({ balance: result.c[0] })
+      })
+    })
+
+    this.setState({currentAddress: this.state.web3.eth.accounts[0]})
+
+  }
+
+  handleSubmitTransfer = (event) => {
+    const { sendAddress, sendAmount, currentAddress } = this.state
+
+    this.state.munCoinInstance.transfer(sendAddress, sendAmount, {from: currentAddress})
+    this.clearForm()
+
+    event.preventDefault()
+  }
+
+  clearForm() {
+    const sendAddress = null
+    const sendAmount = null
+
+    this.setState({ sendAddress, sendAmount })
+  }
+
+  handleInputChange = event => {
+    const { name, value } = event.target
+
+    this.setState({ [name]: value })
+  }
+
+  render() {
+    const { currentAddress, balance, sendAddress, sendAmount } = this.state
+
+    return (
+      <div className="App">
+        <Navigation />
+        <Main currentAddress={currentAddress} balance={balance} />
+          <Form handleSubmit={ this.handleSubmitTransfer } >
+                <Input name="sendAmount" value={sendAmount} handleChange={this.handleInputChange} label="Amount:&nbsp;" type="text" placeholder="0"/>
+                <Input name="sendAddress" value={sendAddress} handleChange={this.handleInputChange} label="To:&nbsp;" type="text" placeholder="0x..." />
+                <Submit type='submit' value='Send' />
+          </Form>
```

Comment:
Hey Jeremy, this is great thanks! I just looked ran it and everything is working perfectly. The `pollWeb3` works better and is neater than the hacky thing I tried before. The alignment of the `Form` was out of the container so I just wrapped it in `Main`:

```javascript
+        <Main currentAddress={currentAddress} balance={balance} >
           <Form handleSubmit={ this.handleSubmitTransfer } >
                 <Input name="sendAmount" value={sendAmount} handleChange={this.handleInputChange} label="Amount:&nbsp;" type="text" placeholder="0"/>
                 <Input name="sendAddress" value={sendAddress} handleChange={this.handleInputChange} label="To:&nbsp;" type="text" placeholder="0x..." />
                 <Submit type='submit' value='Send' />
           </Form>
+       </Main>
```